gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java

```diff
@@ -389,6 +487,12 @@ public void closeProviders() {
   public void setViewMode(Mode viewMode) {
     mapControlsVisibility.postValue(viewMode == Mode.DEFAULT ? VISIBLE : GONE);
     moveFeaturesVisibility.postValue(viewMode == Mode.REPOSITION ? VISIBLE : GONE);
+    addPolygonVisible.postValue(viewMode == Mode.DRAW_POLYGON);
+  }
+
+  private void updatePolygonDrawing(PolygonDrawing polygonDrawing) {
+    addPolygonPoints.postValue(polygonDrawing == PolygonDrawing.STARTED);
```

Comment:
`addPolygonPoints` is not used by anyone. Why can't we clean this?

This would also mean that we can remove the PolygonDrawing enum too

Reply:
ENUM needs to be there as that hold the state of the PolygonDrawing is in.

Reply:
But the state is not being used at any place. Or did I miss something here?

Reply:
function was missing in layout added it there.

Reply:
This name is misleading as well. What does it mean to "update polygon drawing"? What is a "PolygonDrawing"? Please use names that clearly and accurately describe the thing (class, arg) and action (method name) being represented.

Reply:
Also, rather than posting a value to a `MutableLiveData`, please use reactive streams `map` operator to derive the new stream directly as we do elsewhere.

Reply:
i am using the same approach which was used for setViewMode https://github.com/google/ground-android/blob/master/gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java#L387

Reply:
PolygonDrawing was used as the Enum tells the state of PolygonDrawing whether it's started or completed.

Reply:
changed to updateDrawingState in PolygonViewModel.